# Panteones

Nos interesa:
* Panteones con fosas "pantefos"
* Registro de información "panteinf"
* Contacto "panteon" *



In [1]:
import pandas as pd
from IPython.display import display
import matplotlib.pyplot as plt

f_dic = "diccionario_de_datos\diccionario_de_datos_{}_ricpzmp2020.csv"
f_con = "conjunto_datos\{}_ricpzmp2020.csv"
f_cat = "catalogos\{}_ricpzmp2020.csv"

keys = ['zonametr_a', 'entidad_a', 'ubicageo_a', 'nombre1', 'localida_a']

In [2]:
def sustituir_catalogo(df, ndir, nvar):
    s = ndir+f_cat.format(nvar)
    dic = pd.read_csv(s, encoding='latin1')
    dic = dict(zip(dic[nvar], dic['descrip']))
    df[nvar].replace(dic, inplace=True)
    return df

def read_and_clean_csv(ndir, nset):
    # Cargar
    s = ndir+f_con.format(nset)
    print(s)
    df = pd.read_csv(s, encoding='latin1')
    # Datos del diccionario
    s = ndir+f_dic.format(nset)
    print(s)
    dic_col = pd.read_csv(s, encoding='latin1')
    dic_col = dict(zip(dic_col['COLUMNA'], dic_col['DESCRIPCION']))
    # Sustituir valores de catálogo
    for col in df.columns:
        if col not in keys: #catálogo
            try: df = sustituir_catalogo(df, ndir, col)
            except FileNotFoundError: print("\tNo hay catálogo para", col)
            # columna del diccionario
            df = df.rename(columns={col:dic_col[col]})
    return df

## Datos de contacto de panteones

In [3]:
ndir = "panteones_ricpzmp2020_csv/"
nset = "panteon"
nvar = "rensnn1"

print( "PANTEONES" )
df_pan = read_and_clean_csv(ndir, nset)
print(df_pan.shape)
df_pan.tail()

PANTEONES
panteones_ricpzmp2020_csv/conjunto_datos\panteon_ricpzmp2020.csv
panteones_ricpzmp2020_csv/diccionario_de_datos\diccionario_de_datos_panteon_ricpzmp2020.csv
	No hay catálogo para nombre2
	No hay catálogo para correo1
	No hay catálogo para telefo1
(3958, 10)


zonametr_a  entidad_a  ubicageo_a                      nombre1  \
3953        3201         32       32050        VETAGRANDE (SILENCIO)   
3954        3201         32       32056           PANTEÓN DE HERRERA   
3955        3201         32       32056  PANTEÓN JARDÍN DEL RECUERDO   
3956        3201         32       32056             PANTEÓN PURÍSIMA   
3957        3201         32       32057            PANTEÓN MUNICIPAL   

      localida_a Nombre de la localidad  \
3953           1             VETAGRANDE   
3954           1              ZACATECAS   
3955           1              ZACATECAS   
3956           1              ZACATECAS   
3957           1               TRANCOSO   

             Tipo de esquema de prestación de servicios  \
3953  Prestación directa, por parte de alguna instit...   
3954  Prestación directa, por parte de alguna instit...   
3955  Prestación directa, por parte de alguna instit...   
3956  Prestación directa, por parte de alguna instit...   
3957  Prestación directa, por parte de alguna instit...   

     Condición de operación de enero a junio de 2020  \
3953                                              Sí   
3954                                              Sí   
3955                                              Sí   
3956                                              Sí   
3957                                              Sí   

            Correo electrónico de contacto Número telefónico de contacto  
3953       VETAGRANDE2018-2021@HOTMAIL.COM                    4929856020  
3954    PANTEONPURISIMAHERRERA@HOTMAIL.COM                    4929210118  
3955  JOSEMANUELGUTIERREZORTIZ@HOTMAIL.COM                    4921730610  
3956    PANTEONPURISIMAHERRERA@HOTMAIL.COM                    4929210118  
3957          MARTINJASSO.1110@HOTMAIL.COM                    4929312009

## Datos de fosas

In [4]:
ndir = "panteones_operacion_ricpzmp2020_csv/"
nset = "pantefos"

print( "FOSAS" )
df_fos = read_and_clean_csv(ndir, nset)
print(df_fos.shape)
df_fos.tail()

FOSAS
panteones_operacion_ricpzmp2020_csv/conjunto_datos\pantefos_ricpzmp2020.csv
panteones_operacion_ricpzmp2020_csv/diccionario_de_datos\diccionario_de_datos_pantefos_ricpzmp2020.csv
	No hay catálogo para fosas1
	No hay catálogo para fosas2
(1431, 9)


zonametr_a  entidad_a  ubicageo_a                      nombre1  \
1426        3201         32       32050        VETAGRANDE (SILENCIO)   
1427        3201         32       32056           PANTEÓN DE HERRERA   
1428        3201         32       32056  PANTEÓN JARDÍN DEL RECUERDO   
1429        3201         32       32056             PANTEÓN PURÍSIMA   
1430        3201         32       32057            PANTEÓN MUNICIPAL   

     Existencia de fosas comunes Fosas comunes en operación  \
1426                          No                        NaN   
1427                          No                        NaN   
1428                          No                        NaN   
1429                          No                        NaN   
1430                          No                        NaN   

     Fosas comunes cerradas  \
1426                    NaN   
1427                    NaN   
1428                    NaN   
1429                    NaN   
1430                    NaN   

     Ampliación del espacio de las fosas comunes ya existentes o creación de nuevas fosas comunes de enero a junio de 2020  \
1426                                          No aplica                                                                      
1427                                          No aplica                                                                      
1428                                          No aplica                                                                      
1429                                          No aplica                                                                      
1430                                          No aplica                                                                      

      localida_a  
1426           1  
1427           1  
1428           1  
1429           1  
1430           1

Agregar datos de contacto


NOTA:  Hay 3958 pero solo 1431 en la tabla de Fosas Comunes. 
Solo se agrega datos alos panteones de "pantefos", que suponemos son los que contestaron la pregunta de fosas comunes.


In [5]:
df_fos = pd.merge(df_fos, df_pan, how="left", on=keys)
df_fos

zonametr_a  entidad_a  ubicageo_a                      nombre1  \
0            101          1        1001                     DE JESÚS   
1            101          1        1001                  LA ASUNCIÓN   
2            101          1        1001                      LA CRUZ   
3            101          1        1001                     LA SALUD   
4            101          1        1001                  LOS ÁNGELES   
...          ...        ...         ...                          ...   
1428        3201         32       32050        VETAGRANDE (SILENCIO)   
1429        3201         32       32056           PANTEÓN DE HERRERA   
1430        3201         32       32056  PANTEÓN JARDÍN DEL RECUERDO   
1431        3201         32       32056             PANTEÓN PURÍSIMA   
1432        3201         32       32057            PANTEÓN MUNICIPAL   

     Existencia de fosas comunes Fosas comunes en operación  \
0                             No                        NaN   
1                             No                        NaN   
2                             No                        NaN   
3                             No                        NaN   
4                             No                        NaN   
...                          ...                        ...   
1428                          No                        NaN   
1429                          No                        NaN   
1430                          No                        NaN   
1431                          No                        NaN   
1432                          No                        NaN   

     Fosas comunes cerradas  \
0                       NaN   
1                       NaN   
2                       NaN   
3                       NaN   
4                       NaN   
...                     ...   
1428                    NaN   
1429                    NaN   
1430                    NaN   
1431                    NaN   
1432                    NaN   

     Ampliación del espacio de las fosas comunes ya existentes o creación de nuevas fosas comunes de enero a junio de 2020  \
0                                             No aplica                                                                      
1                                             No aplica                                                                      
2                                             No aplica                                                                      
3                                             No aplica                                                                      
4                                             No aplica                                                                      
...                                                 ...                                                                      
1428                                          No aplica                                                                      
1429                                          No aplica                                                                      
1430                                          No aplica                                                                      
1431                                          No aplica                                                                      
1432                                          No aplica                                                                      

      localida_a  Nombre de la localidad  \
0            315  PEÑUELAS (EL CIENEGAL)   
1              1          AGUASCALIENTES   
2              1          AGUASCALIENTES   
3              1          AGUASCALIENTES   
4              1          AGUASCALIENTES   
...          ...                     ...   
1428           1              VETAGRANDE   
1429           1               ZACATECAS   
1430           1               ZACATECAS   
1431           1               ZACATECAS   
1432          

Panteones donde no hay información de fosas en el conteo del INEGI

In [6]:
df_desconocido = df_fos[df_fos['Existencia de fosas comunes'].isin(["Información no disponible", "No se sabe"])]

for col in keys:
    print(col)
    try: df_desconocido = sustituir_catalogo(df_desconocido, ndir, col)
    except: print("\tNo hay catálogo para", col)

df_desconocido.to_csv("PanteonesDatosFosasDesconocido.csv")
df_desconocido.tail()

zonametr_a
entidad_a
ubicageo_a
	No hay catálogo para ubicageo_a
nombre1
	No hay catálogo para nombre1
localida_a
	No hay catálogo para localida_a


/home/esther/anaconda3/envs/panteones/lib/python3.7/site-packages/pandas/core/series.py:4569: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  method=method,


zonametr_a                        entidad_a  ubicageo_a  \
1227  Tlaxcala-Apizaco                         Tlaxcala       29033   
1228  Tlaxcala-Apizaco                         Tlaxcala       29033   
1252  Tlaxcala-Apizaco                         Tlaxcala       29048   
1254  Tlaxcala-Apizaco                         Tlaxcala       29050   
1292        Minatitlán  Veracruz de Ignacio de la Llave       30089   

                         nombre1 Existencia de fosas comunes  \
1227  PANTEÓN TIZATLAN COMUNIDAD   Información no disponible   
1228  PANTEÓN TIZATLAN COMUNIDAD   Información no disponible   
1252               NO DISPONIBLE   Información no disponible   
1254               NO DISPONIBLE   Información no disponible   
1292           PANTEÓN MUNICIPAL                  No se sabe   

     Fosas comunes en operación Fosas comunes cerradas  \
1227                         ND                     ND   
1228                         ND                     ND   
1252                         ND                     ND   
1254                         ND                     ND   
1292                        NSS                    NSS   

     Ampliación del espacio de las fosas comunes ya existentes o creación de nuevas fosas comunes de enero a junio de 2020  \
1227                          Información no disponible                                                                      
1228                          Información no disponible                                                                      
1252                          Información no disponible                                                                      
1254                          Información no disponible                                                                      
1292             No cuenta con elementos para responder                                                                      

      localida_a Nombre de la localidad  \
1227          11   SAN ESTEBAN TIZATLÁN   
1228          11   SAN ESTEBAN TIZATLÁN   
1252        9998          NO DISPONIBLE   
1254        9998          NO DISPONIBLE   
1292           1    JÁLTIPAN DE MORELOS   

             Tipo de esquema de prestación de servicios  \
1227  Prestación directa, por parte de alguna instit...   
1228  Prestación directa, por parte de alguna instit...   
1252                          Información no disponible   
1254                          Información no disponible   
1292  Prestación directa, por parte de alguna instit...   

     Condición de operación de enero a junio de 2020  \
1227                                              Sí   
1228                                              Sí   
1252                       Información no disponible   
1254                       Información no disponible   
1292                                              Sí   

            Correo electrónico de contacto Número telefónico de contacto  
1227                         NO DISPONIBLE                 NO DISPONIBLE  
1228                         NO DISPONIBLE                 NO DISPONIBLE  
1252                         NO DISPONIBLE     INFORMACIÓN NO DISPONIBLE  
1254                         NO DISPONIBLE     INFORMACIÓN NO DISPONIBLE  
1292  REGIDURIASEGUNDAJALTIPAN@HOTMAIL.COM                    9221299710

Panteones con fosas comunes (respondieron Sí)

In [7]:
df_fos = df_fos[df_fos['Existencia de fosas comunes'].isin(["Sí"])]
df_fos

zonametr_a  entidad_a  ubicageo_a                            nombre1  \
21           201          2        2001              PANTEÓN MUNICIPAL # 4   
23           202          2        2002  PANTEÓN MUNICIPAL NO. 3 CENTINELA   
25           203          2        2003             PANTEON MUNICIPAL NO.1   
26           203          2        2003             PANTEON MUNICIPAL NO.2   
27           203          2        2004                              TRECE   
...          ...        ...         ...                                ...   
1415        3101         31       31101               CEMENTERIO DE XTEPÉN   
1416        3101         31       31101            CEMENTERIO DE YAXCOPOIL   
1417        3101         31       31101           CEMENTERIO SANTA MATILDE   
1419        3201         32       32017         PANTEÓN JARDINES GUADALUPE   
1420        3201         32       32017                 PANTEÓN SAN ISIDRO   

     Existencia de fosas comunes Fosas comunes en operación  \
21                            Sí                          9   
23                            Sí                         36   
25                            Sí                          0   
26                            Sí                          3   
27                            Sí                        298   
...                          ...                        ...   
1415                          Sí                          1   
1416                          Sí                          1   
1417                          Sí                          2   
1419                          Sí                          1   
1420                          Sí                          1   

     Fosas comunes cerradas  \
21                       16   
23                       36   
25                       50   
26                       21   
27                        0   
...                     ...   
1415                      0   
1416                      0   
1417                      0   
1419                      0   
1420                      0   

     Ampliación del espacio de las fosas comunes ya existentes o creación de nuevas fosas comunes de enero a junio de 2020  \
21                                                   Sí                                                                      
23                                                   Sí                                                                      
25                                                   No                                                                      
26                                                   Sí                                                                      
27                                                   No                                                                      
...                                                 ...                                                                      
1415                                                 No                                                                      
1416                                                 No                                                                      
1417                                                 No                                                                      
1419                                                 No                                                                      
1420                                                 No                                                                      

      localida_a Nombre de la localidad  \
21             1               ENSENADA   
23             1               MEXICALI   
25             1                 TECATE   
26             1                 TECATE   
27             1                TIJUANA   
...          ...                    ...   
1415          18                 XTEPÉN   
1416          19              YAXCOPOIL   
1417           1                   UMÁN

Agregar los datos de información

In [8]:
ndir = "panteones_operacion_ricpzmp2020_csv/"
nset = "panteinf"

print( "INFORMACION" )
df_inf = read_and_clean_csv(ndir, nset)
print(df_inf.shape)
df_inf.tail()

INFORMACION
panteones_operacion_ricpzmp2020_csv/conjunto_datos\panteinf_ricpzmp2020.csv
panteones_operacion_ricpzmp2020_csv/diccionario_de_datos\diccionario_de_datos_panteinf_ricpzmp2020.csv
	No hay catálogo para sisinf1
	No hay catálogo para sisinf2
	No hay catálogo para sisinf3
	No hay catálogo para sisinf4
(1431, 15)


zonametr_a  entidad_a  ubicageo_a                      nombre1  \
1426        3201         32       32050        VETAGRANDE (SILENCIO)   
1427        3201         32       32056           PANTEÓN DE HERRERA   
1428        3201         32       32056  PANTEÓN JARDÍN DEL RECUERDO   
1429        3201         32       32056             PANTEÓN PURÍSIMA   
1430        3201         32       32057            PANTEÓN MUNICIPAL   

      Sistemas de información - Sobre inhumaciones  \
1426                                           0.0   
1427                                           1.0   
1428                                           1.0   
1429                                           1.0   
1430                                           1.0   

      Sistemas de información - Sobre cremaciones o incineraciones  \
1426                                                0.0              
1427                                                NaN              
1428                                                NaN              
1429                                                NaN              
1430                                                NaN              

      Sistemas de información - Sobre exhumaciones  \
1426                                           0.0   
1427                                           NaN   
1428                                           NaN   
1429                                           NaN   
1430                                           NaN   

      Sistemas de información - Sobre reinhumaciones  \
1426                                             0.0   
1427                                             1.0   
1428                                             1.0   
1429                                             1.0   
1430                                             NaN   

     Sistema de información sobre inhumaciones  \
1426                                 No aplica   
1427                  Libro o bitácora (papel)   
1428                           Hoja de cálculo   
1429                  Libro o bitácora (papel)   
1430                  Libro o bitácora (papel)   

     Sistema de información sobre cremaciones o incineraciones  \
1426                                          No aplica          
1427                                          No aplica          
1428                                          No aplica          
1429                                          No aplica          
1430                                          No aplica          

     Sistema de información sobre exhumaciones  \
1426                                 No aplica   
1427                                 No aplica   
1428                                 No aplica   
1429                                 No aplica   
1430                                 No aplica   

     Sistema de información sobre reinhumaciones  \
1426                                   No aplica   
1427                    Libro o bitácora (papel)   
1428                             Hoja de cálculo   
1429                    Libro o bitácora (papel)   
1430                                   No aplica   

     Registro de la causa de muerte - Sobre inhumaciones  \
1426                                          No aplica    
1427                                                 No    
1428                                                 No    
1429                                                 No    
1430                                                 No    

     Registro de la causa de muerte - Sobre cremaciones o incineraciones  \
1426                                          No aplica                    
1427                                          No aplica                    
1428                                          No aplica                    
1429                                          No aplica                    
1430                                          No aplica                    

      localida_a  
1426           1

In [9]:
df_fos = pd.merge(df_fos, df_inf, how="left", on=['zonametr_a', 'entidad_a', 'ubicageo_a', 'nombre1', 'localida_a'])
df_fos

zonametr_a  entidad_a  ubicageo_a                            nombre1  \
0           201          2        2001              PANTEÓN MUNICIPAL # 4   
1           202          2        2002  PANTEÓN MUNICIPAL NO. 3 CENTINELA   
2           203          2        2003             PANTEON MUNICIPAL NO.1   
3           203          2        2003             PANTEON MUNICIPAL NO.2   
4           203          2        2004                              TRECE   
..          ...        ...         ...                                ...   
244        3101         31       31101               CEMENTERIO DE XTEPÉN   
245        3101         31       31101            CEMENTERIO DE YAXCOPOIL   
246        3101         31       31101           CEMENTERIO SANTA MATILDE   
247        3201         32       32017         PANTEÓN JARDINES GUADALUPE   
248        3201         32       32017                 PANTEÓN SAN ISIDRO   

    Existencia de fosas comunes Fosas comunes en operación  \
0                            Sí                          9   
1                            Sí                         36   
2                            Sí                          0   
3                            Sí                          3   
4                            Sí                        298   
..                          ...                        ...   
244                          Sí                          1   
245                          Sí                          1   
246                          Sí                          2   
247                          Sí                          1   
248                          Sí                          1   

    Fosas comunes cerradas  \
0                       16   
1                       36   
2                       50   
3                       21   
4                        0   
..                     ...   
244                      0   
245                      0   
246                      0   
247                      0   
248                      0   

    Ampliación del espacio de las fosas comunes ya existentes o creación de nuevas fosas comunes de enero a junio de 2020  \
0                                                   Sí                                                                      
1                                                   Sí                                                                      
2                                                   No                                                                      
3                                                   Sí                                                                      
4                                                   No                                                                      
..                                                 ...                                                                      
244                                                 No                                                                      
245                                                 No                                                                      
246                                                 No                                                                      
247                                                 No                                                                      
248                                                 No                                                                      

     localida_a Nombre de la localidad  ...  \
0             1               ENSENADA  ...   
1             1               MEXICALI  ...   
2             1                 TECATE  ...   
3             1                 TECATE  ...   
4             1                TIJUANA  ...   
..          ...                    ...  ...   
244          18                 XTEPÉN  ...   
245          19              YAXCOPOIL  ...   
246           1                   UMÁN  ...   
247

Cambiar nombres llaves y guardar

In [10]:
for col in keys:
    print(col)
    try: df_fos = sustituir_catalogo(df_fos, ndir, col)
    except: print("\tNo hay catálogo para", col)
df_fos.to_csv("PanteonesConFosasComunes.csv")
df_fos.tail()

zonametr_a
entidad_a
ubicageo_a
	No hay catálogo para ubicageo_a
nombre1
	No hay catálogo para nombre1
localida_a
	No hay catálogo para localida_a


zonametr_a  entidad_a  ubicageo_a                     nombre1  \
244               Mérida    Yucatán       31101        CEMENTERIO DE XTEPÉN   
245               Mérida    Yucatán       31101     CEMENTERIO DE YAXCOPOIL   
246               Mérida    Yucatán       31101    CEMENTERIO SANTA MATILDE   
247  Zacatecas-Guadalupe  Zacatecas       32017  PANTEÓN JARDINES GUADALUPE   
248  Zacatecas-Guadalupe  Zacatecas       32017          PANTEÓN SAN ISIDRO   

    Existencia de fosas comunes Fosas comunes en operación  \
244                          Sí                          1   
245                          Sí                          1   
246                          Sí                          2   
247                          Sí                          1   
248                          Sí                          1   

    Fosas comunes cerradas  \
244                      0   
245                      0   
246                      0   
247                      0   
248                      0   

    Ampliación del espacio de las fosas comunes ya existentes o creación de nuevas fosas comunes de enero a junio de 2020  \
244                                                 No                                                                      
245                                                 No                                                                      
246                                                 No                                                                      
247                                                 No                                                                      
248                                                 No                                                                      

     localida_a Nombre de la localidad  ...  \
244          18                 XTEPÉN  ...   
245          19              YAXCOPOIL  ...   
246           1                   UMÁN  ...   
247           1              GUADALUPE  ...   
248           1              GUADALUPE  ...   

    Sistemas de información - Sobre inhumaciones  \
244                                          1.0   
245                                          1.0   
246                                          1.0   
247                                          1.0   
248                                          1.0   

    Sistemas de información - Sobre cremaciones o incineraciones  \
244                                                NaN             
245                                                NaN             
246                                                NaN             
247                                                NaN             
248                                                NaN             

    Sistemas de información - Sobre exhumaciones  \
244                                          1.0   
245                                          1.0   
246                                          1.0   
247                                          1.0   
248                                          1.0   

    Sistemas de información - Sobre reinhumaciones  \
244                                            NaN   
245                                            NaN   
246                                            NaN   
247                                            1.0   
248                                            1.0   

     Sistema de información sobre inhumaciones  \
244                   Libro o bitácora (papel)   
245                   Libro o bitácora (papel)   
246                   Libro o bitácora (papel)   
247                   Libro o bitácora (papel)   
248                   Libro o bitácora (papel)   

     Sistema de información sobre cremaciones o incineraciones  \
244                                          No aplica           
245                                          No aplica           
246                                          No aplica           
247                       